lib/src/modules/task.cc

```diff
@@ -240,30 +241,54 @@ static std::string createUrlEndpoint(const lth_jc::JsonContainer& uri) {
 //    the same task file.
 // The downloaded task file's permissions will be set to rwx for user and rx for
 // group for non-Windows OSes.
-static void downloadTaskFile(const std::vector<std::string>& master_uris,
-                             lth_curl::client& client,
-                             const fs::path& file_path,
-                             const lth_jc::JsonContainer& uri) {
-    auto url = master_uris[0] + createUrlEndpoint(uri);
-    lth_curl::request req(url);
-    // timeout from connection after one minute, can configure
-    req.connection_timeout(60000);
+//
+// The method returns a tuple (success, err_msg). success is true if the file was downloaded;
+// false otherwise. err_msg contains the most recent http_file_download_exception's error
+// message; it is initially empty.
+static boost::tuple<bool, std::string> downloadTaskFile(const std::vector<std::string>& master_uris,
```

Comment:
You probably actually want `std::tuple` (that's the header you included.

Reply:
Will do, I wasn't sure if we should use boost or std there b/c we use boost more often (i.e. not sure of when to use boost vs. std).